### PR TITLE
feat(policies): add VLA-JEPA

### PR DESCRIPTION
**Pull request by Maximellerbach**
## Adding VLA-JEPA to LeRobot

This supersedes #3262

### Tested the LIBERO model converted to LeRobot safetensors format
```bash
lerobot-eval \
    --policy.path=maximellerbach/VLA-JEPA-LIBERO \
    --env.type=libero \
    --env.task=libero_spatial,libero_object,libero_goal,libero_10 \
    --env.observation_height=224 \
    --env.observation_width=224 \
    --eval.batch_size=5 \
    --eval.n_episodes=10 \
    --env.max_parallel_tasks=4
```

| Suite          | Episodes | Successes | Success Rate |
|----------------|----------|-----------|--------------|
| libero_spatial | 100      | 95        | **95.0%**    |
| libero_object  | 100      | 100       | **100.0%**   |
| libero_goal    | 100      | 98        | **98.0%**    |
| libero_10      | 100      | 93        | **93.0%**    |
| **Overall**    | **400**  | **387**   | **96.5%**   |

Roughly the same as the paper

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`)
- [x] All tests pass locally (`pytest`)
- [x] Documentation updated
- [ ] CI is green

## Reviewer notes

- ~~Weights are currently uploaded to `https://huggingface.co/collections/maximellerbach/vla-jepa` they will be uploaded to `lerobot` org once this PR is ready to be merged~~ --> the weights were uploaded to `lerobot` org

- we saw a side-effect on pi0 tests, investigating

### CODE REVIEW

**Review by wadeKeith — COMMENTED** [review]
Withdrawn; this review was too cursory.

**Comment by Maximellerbach** [comment]
need to double check everything from the [bring your own policies docs](https://huggingface.co/docs/lerobot/main/en/bring_your_own_policies) before marking this PR as ready for review

- [x] `MyPolicy` and `MyPolicyConfig` cover the surface above; `__init_subclass__` accepts the class.
- [x] `factory.py` and `policies/__init__.py` are wired (lazy imports for modeling).
- [x] `make_my_policy_pre_post_processors` follows the naming convention.
- [x] Optional deps live behind a `[project.optional-dependencies]` extra and the `TYPE_CHECKING + require_package` guard.
- [x] `tests/policies/` updated; backward-compat artifact committed & policy-specific tests.
- [x] `src/lerobot/policies/<name>/README.md` symlinked into `docs/source/policy_<name>_README.md`; user-facing `docs/source/<name>.mdx` written and added to `_toctree.yml`.
- [x] At least one reproducible benchmark eval in the policy MDX with a published checkpoint (sim benchmark, or real-robot dataset + checkpoint).

File: tests/policies/vla_jepa/test_vla_jepa.py [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+    BATCH_SIZE,
+    EXPECTED_ACTION_CHUNK_SHAPE,
+    EXPECTED_SELECT_ACTION_SHAPE,
+    N_ACTION_STEPS,
+    STATE_DIM,
+    make_config,
+    make_inference_batch,
+    make_train_batch,
+    set_seed_all,
+)
```
[the remark this answers is not in the document]
Comment (Maximellerbach) [reply tests/policies/vla_jepa/test_vla_jepa.py:34]: tests run fine

File: tests/policies/vla_jepa/test_action_head.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+from conftest import (
+    ACTION_DIM,
+    ACTION_HORIZON,
+    BATCH_SIZE,
+    QWEN_HIDDEN_SIZE,
+    STATE_DIM,
+    make_config,
+    set_seed_all,
+)  # noqa: E402
+
```
[the remark this answers is not in the document]
Comment (Maximellerbach) [reply tests/policies/vla_jepa/test_action_head.py:19]: tests run fine

File: src/lerobot/policies/vla_jepa/action_head.py [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+        time_emb = self.pos_encoding(timesteps).to(dtype=action_emb.dtype)
+        return self.layer3(swish(self.layer2(torch.cat([action_emb, time_emb], dim=-1))))
+
+
+class TimestepEncoder(nn.Module):
+    def __init__(self, embedding_dim: int):
+        super().__init__()
+        self.time_proj = Timesteps(num_channels=256, flip_sin_to_cos=True, downscale_freq_shift=1)
+        self.timestep_embedder = TimestepEmbedding(in_channels=256, time_embed_dim=embedding_dim)
+
```
[the remark this answers is not in the document]
Comment (Maximellerbach) [reply src/lerobot/policies/vla_jepa/action_head.py:86]: adressed in [dd13eda](https://github.com/huggingface/lerobot/pull/3568/commits/dd13eda002bc56ffd88bff1e87f0e005ef4203fb)

File: src/lerobot/policies/vla_jepa/modeling_vla_jepa.py [hunk 4]
```diff
@@ -0,0 +1,608 @@
...
+    # ---- LeRobot Policy Interface ----
+
+    def forward(self, batch: dict[str, Tensor]) -> tuple[Tensor, dict]:
+        """LeRobot train forward: convert → native forward → aggregate losses."""
+        examples = self._prepare_model_inputs(batch)
+        native_output = self.model.forward(examples)
+
+        total_loss = native_output.get("action_loss", torch.tensor(0.0)) + native_output.get(
+            "wm_loss", torch.tensor(0.0)
+        )
```
[the remark this answers is not in the document]
Comment (Maximellerbach) [reply src/lerobot/policies/vla_jepa/modeling_vla_jepa.py]: adressed in [c01a00a](https://github.com/huggingface/lerobot/pull/3568/commits/c01a00a972ac7d6bbb4dbcdde60d89a0a9a2ed1f)

**Comment by Maximellerbach** [comment]
@claude review this

**Comment by Maximellerbach** [comment]
The models weights were just posted here [https://huggingface.co/collections/lerobot/vla-jepa](https://huggingface.co/collections/lerobot/vla-jepa)

File: src/lerobot/policies/vla_jepa/__init__.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from .configuration_vla_jepa import VLAJEPAConfig
```
Comment (s1lent4gnt) [inline src/lerobot/policies/vla_jepa/__init__.py:15]: Could you add license here.

File: src/lerobot/policies/vla_jepa/processor_vla_jepa.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+        PolicyProcessorPipeline[PolicyAction, PolicyAction](
+            steps=output_steps,
+            name=POLICY_POSTPROCESSOR_DEFAULT_NAME,
+            to_transition=policy_action_to_transition,
+            to_output=transition_to_policy_action,
+        ),
+    )
+
+
+@ProcessorStepRegistry.register(name="vla_jepa_new_line_processor")
```
Comment (s1lent4gnt) [inline src/lerobot/policies/vla_jepa/processor_vla_jepa.py]: Is this processor needed?

File: src/lerobot/policies/vla_jepa/__init__.py (hunk 5 above)
Comment (Maximellerbach) [reply src/lerobot/policies/vla_jepa/__init__.py:15]: resolved in[e1852db](https://github.com/huggingface/lerobot/pull/3568/commits/e1852db71a0a86a1ae8c7e71545b481f7beba29c)

File: src/lerobot/policies/vla_jepa/processor_vla_jepa.py (hunk 6 above)
Comment (Maximellerbach) [reply src/lerobot/policies/vla_jepa/processor_vla_jepa.py]: nope, will remove it

Comment (Maximellerbach) [reply src/lerobot/policies/vla_jepa/processor_vla_jepa.py]: adressed in [a98c56](https://github.com/huggingface/lerobot/pull/3568/commits/7a98c56ce468e23e54c58f641357a3e559f9c473)

File: src/lerobot/policies/vla_jepa/action_head.py [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+    class ConfigMixin:  # type: ignore[no-redef]
+        pass
+
+    register_to_config = lambda f: f  # noqa: E731
+    Attention = FeedForward = TimestepEmbedding = Timesteps = None
+
+from .configuration_vla_jepa import VLAJEPAConfig
+
+
+def swish(x: torch.Tensor) -> torch.Tensor:
```
Comment (s1lent4gnt) [inline src/lerobot/policies/vla_jepa/action_head.py]: We can remove this in favor of F.silu

File: src/lerobot/policies/vla_jepa/processor_vla_jepa.py [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    required when the model was trained with gripper in identity (mask=False)
+    space where 0=open and 1=close.
+    """
+
+    def __call__(self, transition: EnvTransition) -> EnvTransition:
+        action = transition.get(TransitionKey.ACTION)
+        if action is not None and action.shape[-1] >= 7:
+            transition = dict(transition)
+            a = action.clone()
+            a[..., 6] = (a[..., 6] >= 0.5).float()
```
Comment (s1lent4gnt) [inline src/lerobot/policies/vla_jepa/processor_vla_jepa.py]: Hardcoded gripper index 6, better make it configurable

File: src/lerobot/policies/vla_jepa/modeling_vla_jepa.py [hunk 9]
```diff
@@ -0,0 +1,629 @@
...
+
+        self.model = VLAJEPAModel(config)
+        self.reset()
+
+    def reset(self) -> None:
+        self._queues = {ACTION: deque(maxlen=self.config.n_action_steps)}
+
+    # ---- Format Conversion: LeRobot → Native ----
+
+    def _prepare_model_inputs(self, batch: dict[str, Tensor]) -> list[dict]:
```
Comment (s1lent4gnt) [inline src/lerobot/policies/vla_jepa/modeling_vla_jepa.py:422]: Can we remove the GPU→CPU→GPU round-trips

Comment (Maximellerbach) [reply src/lerobot/policies/vla_jepa/modeling_vla_jepa.py:422]: will issue this one in a follow-up PR if possible to remove

File: src/lerobot/policies/vla_jepa/processor_vla_jepa.py (hunk 8 above)
Comment (Maximellerbach) [reply src/lerobot/policies/vla_jepa/processor_vla_jepa.py]: adressed in [61d82a5](https://github.com/huggingface/lerobot/pull/3568/commits/61d82a57735e5d1748351e6c04c1f81113718b59)

Also updated the configs of the already uploaded models so that there is no draccus error when loaded etc.

File: src/lerobot/policies/vla_jepa/README.md [hunk 10]
```diff
@@ -0,0 +1 @@
+/home/maxime/github/robots/lerobot/docs/source/policy_vla_jepa_README.md
```
Comment (s1lent4gnt) [inline src/lerobot/policies/vla_jepa/README.md]: Could you make this relative?

Comment (Maximellerbach) [reply src/lerobot/policies/vla_jepa/README.md]: oh yeah right sorry about this, changing it rn

File: src/lerobot/policies/vla_jepa/action_head.py (hunk 7 above)
Comment (Maximellerbach) [reply src/lerobot/policies/vla_jepa/action_head.py]: resolved in [d8ed30d](https://github.com/huggingface/lerobot/pull/3568/commits/d8ed30d58cf43abe38140f6178443a818c2d925b)

File: src/lerobot/policies/vla_jepa/README.md (hunk 10 above)
Comment (Maximellerbach) [reply src/lerobot/policies/vla_jepa/README.md]: resolved in [ac9128b](https://github.com/huggingface/lerobot/pull/3568/commits/ac9128b3194efbe8869aa2707e90f50f32da3bd6)

File: src/lerobot/policies/vla_jepa/modeling_vla_jepa.py [hunk 11]
```diff
@@ -0,0 +1,629 @@
...
+        num_action_prompt_steps = self.config.num_video_frames // _tubelet_size - 1
+        self.replace_prompt = "".join(
+            token * self.config.num_action_tokens_per_timestep
+            for token in self.action_tokens[:num_action_prompt_steps]
+        )
+        self.embodied_replace_prompt = (
+            self.config.embodied_action_token * self.config.num_embodied_action_tokens_per_instruction
+        )
+
+    def _qwen_last_decoder_hidden(self, qwen_inputs: dict[str, torch.Tensor]) -> torch.Tensor:
```
Comment (Maximellerbach) [inline src/lerobot/policies/vla_jepa/modeling_vla_jepa.py:133]: quick note, the issue with the initial reproducing of the libero result was because of a transformers breaking change. the "why" and how it was fixed to align with previous versions is documented in this function

**Review by s1lent4gnt — DISMISSED** [review]
VLA with World Model auxiliary training loss, very nice work @Maximellerbach !!
LGTM!
